web/src/pages/api/post/create.ts

```diff
@@ -57,24 +57,31 @@ export default async function create(req: Request, res: NextApiResponse) {
   // console.log(DiffusionModelPrediction[0]);
   // return(res.json({'image': DiffusionModelPrediction[0]}));
 
-  if (data.status == 406) {
-    return res.status(403).json({ error: "NSFW CONTENT REJECTED." });
+  if (sdRes.status == 406) {
+    return res.status(403).json({ error: "NSFW prompt rejected" });
+  } else if (!sdRes.ok) {
+    return res.status(500).json({ error: "Error while creating image" });
   }
   //response data into json
 
-  const resData = (await data.json()) as ResponseData;
+  const resData = (await sdRes.json()) as ResponseData;
 
-  const post = await prisma.post.create({
-    data: {
-      prompt: prompt,
-      imageURL: resData.image,
-      authorId: userId,
-    },
-  });
+  const post = await prisma.post
+    .create({
+      data: {
+        prompt: prompt,
+        imageURL: resData.image,
+        authorId: userId,
+      },
+    })
+    .catch((e: Error) => {
+      console.error(e.message);
+      return res.status(500).json({ error: "Internal database error" });
+    });
 
   return res.json({
-    postId: post.id,
-    prompt: post.prompt,
-    image: post.imageURL,
+    postId: post?.id,
+    prompt: post?.prompt,
+    image: post?.imageURL,
```

Comment:
I don't think this is what we want - you'll just send JSON with `undefined` in it if the creation fails.

Reply:
I'll now return this inside of a .then to prevent the type issue